### PR TITLE
fix cpu broadcast

### PR DESCRIPTION
**Pull request by jacquesqiao**
CE通过 http://yq01-paddle-online.epc.baidu.com:8098/commit/details?commit=3bccc1e6e275412f30baf5a0c5698eb307f90252

### CODE REVIEW

**Review by chengduoZH — DISMISSED** [review]

File: paddle/fluid/framework/details/build_strategy.cc [hunk 1]
```diff
@@ -133,12 +133,15 @@ class ParallelExecutorPassBuilder : public ir::PassBuilder {
   void AppendMultiDevPass(const BuildStrategy &strategy) {
     ir::Pass *multi_devices_pass;
     if (strategy_.is_distribution_) {
+      VLOG(3) << "multi device dist train mode";
```
Comment (Yancey0623) [inline paddle/fluid/framework/details/build_strategy.cc]: multi devices pserver mode;
multi devices collective mode with allreduce;
multi deivces collective mode with reduce?

File: python/paddle/fluid/compiler.py [hunk 2]
```diff
@@ -34,6 +35,15 @@ def _place_obj(place):
     return p
 
 
+def _is_pserver_mode(main_program):
```
Comment (Yancey0623) [inline python/paddle/fluid/compiler.py]: Duplicated with parallel_executor.py#is_pserver_mode()

**Review by Yancey0623 — APPROVED** [review]
lgtm

File: python/paddle/fluid/parallel_executor.py [hunk 3]
```diff
@@ -140,7 +131,7 @@ def __init__(self,
         # FIXME(zcd): is_distribution_ is a temporary field, because in pserver mode,
         # num_trainers is 1, so the current fields of build_strategy doesn't tell if
         # it's distributed model.
-        build_strategy.is_distribution = _is_pserver_mode(
+        build_strategy.is_distribution = framework.is_pserver_mode(
```
Comment (chengduoZH) [inline python/paddle/fluid/parallel_executor.py:134]: Not sure is it appropriate that moving `is_pserver_mode` to `framework` model.
In fact, `parallel_executor.py` will be removed in the near future.

**Pull request merged**